internal/statesync/streamer.go

```diff
@@ -0,0 +1,79 @@
+package statesync
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/kwilteam/kwil-db/core/log"
+)
+
+// Utility to stream chunks of a snapshot
+type Streamer struct {
+	log               log.Logger
+	numChunks         uint32
+	files             []string
+	currentChunk      *os.File
+	currentChunkIndex uint32
+}
+
+func NewStreamer(numChunks uint32, chunkDir string, logger log.Logger) *Streamer {
+	files := make([]string, numChunks)
+	for i := uint32(0); i < numChunks; i++ {
+		file := filepath.Join(chunkDir, fmt.Sprintf("chunk-%d.sql.gz", i))
+		files[i] = file
+	}
+
+	return &Streamer{
+		log:       logger,
+		numChunks: numChunks,
+		files:     files,
+	}
+}
+
+func (s *Streamer) Next() error {
+	if s.currentChunk != nil {
+		s.currentChunk.Close()
+	}
+
+	if s.currentChunkIndex >= s.numChunks {
+		return io.EOF // no more chunks to stream
+	}
+
+	file, err := os.Open(s.files[s.currentChunkIndex])
+	if err != nil {
+		return fmt.Errorf("failed to open chunk file: %w", err)
+	}
+
+	s.currentChunk = file
+	s.currentChunkIndex++
+
+	return nil
+}
+
+func (s *Streamer) Close() error {
+	if s.currentChunk != nil {
+		s.currentChunk.Close()
+	}
+
+	return nil
+}
+
+func (s *Streamer) Read(p []byte) (n int, err error) {
```

Comment:
Some better documentation on the exported structs and methods here could be helpful.